src/test/java/org/broadinstitute/hellbender/tools/GenomicsDBImportIntegrationTest.java

```diff
@@ -26,68 +29,150 @@
+        final String hg00096 = largeFileTestDir + "gvcfs/HG00096.g.vcf.gz";
+        final String hg00268 = largeFileTestDir + "gvcfs/HG00268.g.vcf.gz";
+        final String na19625 = largeFileTestDir + "gvcfs/NA19625.g.vcf.gz";
+        final String combined = largeFileTestDir + "gvcfs/combined.gatk3.7.g.vcf.gz";
+        final SimpleInterval interval = new SimpleInterval("chr20", 17960187, 17981445);
+
+        testGenomicsDBImportWithZeroBufferSize(Arrays.asList(hg00096, hg00268, na19625), interval, combined);
+    }
+
+    private void testGenomicsDBImporter(final List<String> vcfInputs, final SimpleInterval interval, final String expectedCombinedVCF) throws IOException {
+        final String workspace = createTempDir("genomicsdb-tests-").getAbsolutePath() + "/workspace";
+
+        writeToGenomicsDB(vcfInputs, interval, workspace, 0, false, 0);
+        checkJSONFilesAreWritten(workspace);
+        checkGenomicsDBAgainstExpected(workspace, interval, expectedCombinedVCF);
+    }
+
+    private void testGenomicsDBImporterWithBatchSize(final List<String> vcfInputs, final SimpleInterval interval, final String expectedCombinedVCF, int batchSize) throws IOException {
+        final String workspace = createTempDir("genomicsdb-batchsize-tests-").getAbsolutePath() + "/workspace-" + batchSize;
+
+        writeToGenomicsDB(vcfInputs, interval, workspace, batchSize, false, 0);
+        checkGenomicsDBAgainstExpected(workspace, interval, expectedCombinedVCF);
+    }
+
+    private void testGenomicsDBImportWithZeroBufferSize(final List<String> vcfInputs, final SimpleInterval interval, final String expectedCombinedVCF) throws IOException {
+        final String workspace = createTempDir("genomicsdb-buffersize-tests-").getAbsolutePath() + "/workspace";
+
+
+        writeToGenomicsDB(vcfInputs, interval, workspace, 0, true, 0);
+        checkJSONFilesAreWritten(workspace);
+        checkGenomicsDBAgainstExpected(workspace, interval, expectedCombinedVCF);
+
+    }
+
+    private void writeToGenomicsDB(final List<String> vcfInputs, final SimpleInterval interval, final String workspace, int batchSize, final Boolean useBufferSize, int bufferSizePerSample) {
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addArgument("genomicsDBWorkspace", workspace);
+        args.addArgument("L", IntervalUtils.locatableToString(interval));
+        vcfInputs.forEach(vcf -> args.addArgument("V", vcf));
+        args.addArgument("batchSize", String.valueOf(batchSize));
+
+        if (useBufferSize)
+            args.addArgument("genomicsDBVCFBufferSize", String.valueOf(bufferSizePerSample));
+
+        runCommandLine(args);
+    }
+
+    private void checkJSONFilesAreWritten(String workspace) {
+        Assert.assertTrue(new File(workspace, GenomicsDBConstants.DEFAULT_VIDMAP_FILE_NAME).exists());
+        Assert.assertTrue(new File(workspace, GenomicsDBConstants.DEFAULT_CALLSETMAP_FILE_NAME).exists());
```

Comment:
Inside GenomicsDB we validate the json files. I think checking the content and validity again will just add redundant code here. But, I will add the check back to see whether these files are written. Okay?